app/services/encrypted_redis_struct_storage.rb

```diff
@@ -37,6 +37,20 @@ def store(struct, expires_in: 60)
     payload = struct.as_json
     payload.delete('id')
 
+    utf_8_encode_strs = proc do |value|
+      if value.is_a?(String)
+        value.force_encoding('UTF-8')
```

Comment:
Another option I found was this: https://stackoverflow.com/a/14133998

```suggestion
        value.encode('UTF-8', invalid: :replace, undef: :replace, replace: '?')
```

which would give us something like `HTTP Status 401 ??? Unauthorized` which is also correct, but forcing it seemed to preserve the actual data better. There may be future unencodable errors, but I don't know of any now, so my inclination is to leave this as-is